scripts/download_from_lesswrong.com.py

```diff
@@ -1,2 +1,130 @@
+COLLECTION_SLUG = "rationality"
+
+###
+def safe_dirname(name: str) -> str:
+    return name.replace("/", "|")
+
+
+def write_post(postdir: str, post: dict):
+    metadata = {
+        "id": post["_id"],
+        "author": post["user"]["displayName"],
+        "modified_at": post["modifiedAt"],
+        "url": post["pageUrl"],
+    }
+
+    if "tags" in post and isinstance(post["tags"], list):
+        metadata["tags"] = []
+
+        for tag in post["tags"]:
+            metadata["tags"].append(tag["name"])
+
+    try:
+        log.info("Create data files for a post '%s'", postdir)
+
+        p = Path(postdir, "content.html")
+        p.write_text(post["contents"]["html"])
+
+        p = Path(postdir, "metadata")
+        p.write_text(json.dumps(metadata))
+    except Exception as err:
+        log.critical("Can't create data files for post '%s': %s", postdir, err)
+        sys.exit(err)
+
+    return
+
+
+def main():
+    log.info("Download collection '%s'", COLLECTION_SLUG)
+    collection = lw.download_collection(COLLECTION_SLUG)
+
+    if "error" in collection:
+        log.critical(
+            "Collection '%s' not downloaded: %s", COLLECTION_SLUG, collection["error"]
+        )
+        sys.exit("Collection download error")
+
+    collection = collection["data"]
+
+    p_base = Path(ROOT, collection["title"])
+
+    posts_for_download = {}
+
+    log.debug("Create directories tree at '%s'", p_base.absolute())
+
+    for book_i, book in enumerate(collection["books"]):
+        book_dirname = str(book_i + 1).zfill(2) + "_" + safe_dirname(book["title"])
+        book_p = Path(p_base.absolute(), book_dirname)
+
+        for sequence_i, sequence in enumerate(book["sequences"]):
+            sequence_dirname = (
+                str(sequence_i + 1).zfill(2) + "_" + safe_dirname(sequence["title"])
+            )
+            sequence_p = Path(book_p.absolute(), sequence_dirname)
+
+            post_i = 1
+            if "contents" in sequence and isinstance(sequence["contents"], dict):
+
+                intro_post = {
+                    "title": "Intro",
+                    "contents": {"html": sequence["contents"]["html"]},
+                    "modifiedAt": sequence["contents"]["editedAt"],
+                    "_id": sequence["_id"],
+                    "user": sequence["user"],
+                    "pageUrl": "https://www.lesswrong.com/s/" + sequence["_id"],
+                }
+                post_i = 0
+                sequence["chapters"][0]["posts"].insert(0, intro_post)
+
+            for chapter in sequence["chapters"]:
+
+                for post in chapter["posts"]:
+                    post_dirname = (
+                        str(post_i).zfill(2) + "_" + safe_dirname(post["title"])
+                    )
+                    post_p = Path(sequence_p.absolute(), post_dirname)
+
+                    try:
+                        post_p.mkdir(parents=True, exist_ok=True)
+                    except Exception as err:
```

Comment:
same question about exceptions, perhaps info logging is enough

Reply:
sorry, i didn't catch an idea. you mean, that i don't need to use try/except blocks?

Reply:
Yes, result will be same - script will exit, just need to check that last log message has enough info